core_backend/app/routers/manage_content.py

```diff
@@ -99,51 +145,121 @@ async def delete_content(
     Delete content endpoint
     """
     record = await get_content_from_db(
-        content_id,
+        content_text_id,
         asession,
     )
 
     if not record:
         raise HTTPException(
-            status_code=404, detail=f"Content id `{content_id}` not found"
+            status_code=404, detail=f"Content id `{content_text_id}` not found"
         )
-    await delete_content_from_db(content_id, asession)
+    await delete_content_from_db(content_text_id, record.content_id, asession)
 
 
-@router.get("/{content_id}", response_model=ContentRetrieve)
+@router.get(
+    "/{content_text_id}", response_model=Union[ContentRetrieve, List[ContentRetrieve]]
+)
 async def retrieve_content_by_id(
-    content_id: int,
+    content_text_id: int,
     readonly_access_user: Annotated[
         AuthenticatedUser, Depends(get_current_readonly_user)
     ],
     asession: AsyncSession = Depends(get_async_session),
-) -> ContentRetrieve:
+    language: Optional[str] = None,
+) -> Union[ContentRetrieve, List[ContentRetrieve]]:
     """
     Retrieve content by id endpoint
     """
 
-    record = await get_content_from_db(content_id, asession)
+    content = await get_content_from_db(content_text_id, asession)
 
-    if not record:
+    if not content:
+        raise HTTPException(
+            status_code=404, detail=f"Content id `{content_text_id}` not found"
+        )
+
+    if language:
+        record = await get_content_from_content_id_and_language(
+            content.content_id, int(language), asession
+        )
+        if not record:
+            raise HTTPException(
+                status_code=404,
+                detail=f"""Content `{content_text_id}`
+                with language id `{language}` not found""",
+            )
+        return _convert_record_to_schema(record)
+    else:
+        records = await get_all_languages_version_of_content(
+            content.content_id, asession
+        )
+        return [_convert_record_to_schema(record) for record in records]
+
+
+async def is_content_and_language_valid(
+    content: ContentTextCreate,
+    asession: AsyncSession,
+    is_edit: bool = False,
+    is_updated_language: bool = True,
+) -> bool:
```

Comment:
Are your pre-commit hooks running? I think interrogate should have raised an error for missing docstring. 

Also, I am finding the conditionals are a bit difficult to parse, so maybe listing out all the invalid/valid cases here would be helpful.

Reply:
When you say here you mean on the docstring?